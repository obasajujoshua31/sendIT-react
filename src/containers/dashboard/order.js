import React, { Component } from "react";
import { withRouter } from "react-router-dom";

class Order extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const {
      parcel: {
        parcel_id,
        parcel_name,
        destination,
        pick_up_location,
        sent_on,
        status
      },
      onHandleViewOrder
    } = this.props;
    return (
      <React.Fragment>
        <tr onClick={() => onHandleViewOrder(parcel_id)}>
          <th scope="row">{`#${parcel_id}`}</th>
          <td>{parcel_name}</td>
          <td>{pick_up_location}</td>
          <td>{destination}</td>
          <td>{sent_on}</td>
          <td>{status}</td>
        </tr>
      </React.Fragment>
    );
  }
}

export default withRouter(Order);
