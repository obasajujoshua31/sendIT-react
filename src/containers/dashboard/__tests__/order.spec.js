import React from "react";
import { Order } from "../order";
import { shallow } from "enzyme";
import sinon from "sinon";

const props = {
  onHandleViewOrder: jest.fn(),
  parcel: {
    parcel_id: "",
    parcel_name: "",
    destination: "",
    pick_up_location: "",
    sent_on: "",
    status: ""
  }
};

const wrapper = shallow(<Order {...props} />);
describe("Test for Parcel Order", () => {
  it("should render Dashboard component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onHandleViewOrder", () => {
    wrapper.find("tr").simulate("click");
    expect(wrapper.instance().props.onHandleViewOrder).toBeCalledWith(
      wrapper.instance().props.parcel.parcel_id
    );
  });
});
