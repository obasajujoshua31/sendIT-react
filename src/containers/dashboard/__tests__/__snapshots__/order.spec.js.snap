// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Parcel Order should render Dashboard component 1`] = `
<Fragment>
  <tr
    onClick={[Function]}
  >
    <th
      scope="row"
    >
      #
    </th>
    <td />
    <td />
    <td />
    <td />
    <td />
  </tr>
</Fragment>
`;
